paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewState.kt

```diff
@@ -114,15 +116,18 @@ internal sealed class CustomerSheetViewState(
         val displayDismissConfirmationModal: Boolean = false,
         val bankAccountResult: CollectBankAccountResultInternal?,
         override val cbcEligibility: CardBrandChoiceEligibility,
+        val errorReporter: ErrorReporter,
     ) : CustomerSheetViewState(
         savedPaymentMethods = emptyList(),
         isLiveMode = isLiveMode,
         isProcessing = isProcessing,
         isEditing = false,
         screen = if (isFirstPaymentMethod) {
-            PaymentSheetScreen.AddFirstPaymentMethod
+            PaymentSheetScreen.AddFirstPaymentMethod(interactor = UnsupportedAddPaymentMethodInteractor(errorReporter))
         } else {
-            PaymentSheetScreen.AddAnotherPaymentMethod
+            PaymentSheetScreen.AddAnotherPaymentMethod(
+                interactor = UnsupportedAddPaymentMethodInteractor(errorReporter)
+            )
```

Comment:
I don't really like having this `UnsupportedAddPaymentMethodInteractor` here. It should be really obvious if we ever depend on it, so I think this is fine, but we could also:

- make the interactor nullable on these screens and set it to null here. Then we could just make the Content for the screens be empty if the interactor is null
- Add an enum to PaymentSheetScreen with a value for each screen. We could it use for HeaderTextFactory, PaymentSheetTopBarStateFactory, etc, which don't really depend on the whole screens and just want to know which screen we're using.

Curious if you have an opinion! If I was to add the enum to PaymentSheetScreen, I'd probably prefer to do it in a follow up since it's otherwise unrelated to this PR

Reply:
And the reason you don't need the real one is because the screen isn't used to be displayed, but is only used for the header text?

In that case it seems like it's just not the right abstraction to pass the screen into the header text.